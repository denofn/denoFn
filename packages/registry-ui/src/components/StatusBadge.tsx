import React from "react";
import { css } from "otion";

type StatusBadgeProps = {
  status: boolean;
  statusText: [string, string];
};

export function StatusBadge({ status, statusText }: StatusBadgeProps) {
  const [positive, negative] = statusText;

  return (
    <div
      className={css({
        display: "inline-block",
        borderRadius: "9999px",
        fontWeight: "bold",
        fontSize: ".75rem",
        marginTop: ".75rem",
        padding: ".25rem .5rem",
        textTransform: "uppercase",
        backgroundColor: status ? "rgb(56,178,172)" : "rgb(245,101,101)",
        color: status ? "rgb(35,78,82)" : "white",
      })}
    >
      {status ? positive : negative}
    </div>
  );
}
