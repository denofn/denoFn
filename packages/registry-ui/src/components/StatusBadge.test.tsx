import { render, cleanup } from "@testing-library/react";
import React from "react";

import { StatusBadge } from "./StatusBadge";

afterEach(cleanup);

it("renders with the correct styles", () => {
  const { asFragment } = render(<StatusBadge status statusText={["a", "b"]} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders with the correct styles for falsy status", () => {
  const { asFragment } = render(<StatusBadge status={false} statusText={["a", "b"]} />);
  expect(asFragment()).toMatchSnapshot();
});
