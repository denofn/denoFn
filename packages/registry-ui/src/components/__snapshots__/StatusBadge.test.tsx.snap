// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the correct styles 1`] = `
<DocumentFragment>
  <div
    class="_1r5gb7q _j033lg _1efi8gv _180q8om _1dbsnr2 _r24itg _50zrmy _st1wxx _1y5pc60"
  >
    a
  </div>
</DocumentFragment>
`;

exports[`renders with the correct styles for falsy status 1`] = `
<DocumentFragment>
  <div
    class="_1r5gb7q _j033lg _1efi8gv _180q8om _1dbsnr2 _r24itg _50zrmy _6zs611 _1y5pc60"
  >
    b
  </div>
</DocumentFragment>
`;
